File: PIPromptClases/MyApplication4/app/src/main/java/com/example/myapplication/UsuarioEvento.kt
package com.example.myapplication

import java.time.LocalDateTime

data class UsuarioEvento(
    var usuarioEventoId: Int,
    var usuarioId: Int,
    var eventoId: Int,
    var fechaRegistro: LocalDateTime,
    var asistencia: Boolean
) {
    constructor() : this(
        usuarioEventoId = 0,
        usuarioId = 0,
        eventoId = 0,
        fechaRegistro = LocalDateTime.now(),
        asistencia = false
    )
}

File: frontend-wpf/AplicacionEscritorioWPF/PrototipadoEscritorio/PrototipadoEscritorio/Models/UsuarioEvento.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using CommunityToolkit.Mvvm.ComponentModel;

namespace PrototipadoEscritorio.Models
{
    public partial class UsuarioEvento : ObservableObject
    {
        public int UsuarioEventoId { get; set; } // No reactivo
        public int UsuarioId { get; set; }        // No reactivo
        public int EventoId { get; set; }         // No reactivo
        public DateTime FechaRegistro { get; set; } // No reactivo

        [ObservableProperty]
        private bool asistencia; // Reactivo

        public UsuarioEvento() { }

        public UsuarioEvento(int usuarioEventoId, int usuarioId, int eventoId, DateTime fechaRegistro, bool asistencia)
        {
            UsuarioEventoId = usuarioEventoId;
            UsuarioId = usuarioId;
            EventoId = eventoId;
            FechaRegistro = fechaRegistro;
            Asistencia = asistencia;
        }
    }
}

